cpp/CMakeLists.txt

```diff
@@ -0,0 +1,62 @@
+cmake_minimum_required(VERSION 3.8)
+project(foxsocketpp LANGUAGES CXX)
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED on)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+option(WERROR "Add -Werror flag to build (turns warnings into errors)" ON)
```

Comment:
Remove and just make this always on? :)

Reply:
I've seen this used (and used it myself) in various projects because compilers tend to add more warnings over time. So code that compiles fine with today's llvm 13 may throw warnings with tomorrow's llvm 14. This is a quick unblocker for people who want to use newer/different compilers or different compiler flags without modifying the codebase.